Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SimpleShapeMonteCarloAbsorption.py

```diff
@@ -0,0 +1,311 @@
+                             validator=IntBoundedValidator(1),
+                             doc='Number of wavelengths for calculation')
+
+        self.declareProperty(name='EventsPerPoint', defaultValue=1000,
+                             validator=IntBoundedValidator(0),
+                             doc='Number of neutron events')
+
+        self.declareProperty(name='Interpolation', defaultValue='Linear',
+                             validator=StringListValidator(['Linear', 'CSpline']),
+                             doc='Type of interpolation')
+
+        # -------------------------------------------------------------------------------------------
+
+        # Beam size
+        self.declareProperty(name='BeamHeight', defaultValue=1.0,
+                             validator=FloatBoundedValidator(0.0),
+                             doc='Height of the beam (cm)')
+
+        self.declareProperty(name='BeamWidth', defaultValue=1.0,
+                             validator=FloatBoundedValidator(0.0),
+                             doc='Width of the beam (cm)')
+
+        # -------------------------------------------------------------------------------------------
+
+        # set up shape options
+
+        self.declareProperty(name='Shape', defaultValue='FlatPlate',
+                             validator=StringListValidator(['FlatPlate', 'Cylinder', 'Annulus']),
+                             doc='Geometry of sample environment. Options are: FlatPlate, Cylinder, Annulus')
+
+        flatPlateCondition = VisibleWhenProperty('Shape', PropertyCriterion.IsEqualTo, 'FlatPlate')
+        cylinderCondition = VisibleWhenProperty('Shape', PropertyCriterion.IsEqualTo, 'Cylinder')
+        annulusCondition = VisibleWhenProperty('Shape', PropertyCriterion.IsEqualTo, 'Annulus')
+
+        # height is common to all options
+
+        self.declareProperty(name='Height', defaultValue=0.0,
+                             validator=FloatBoundedValidator(0.0),
+                             doc='Height of the sample environment')
+
+        # flat plate options
+
+        self.declareProperty(name='Width', defaultValue=0.0,
+                             validator=FloatBoundedValidator(0.0),
+                             doc='Width of the FlatPlate sample environment')
+        self.setPropertySettings('Width', flatPlateCondition)
+
+        self.declareProperty(name='Thickness', defaultValue=0.0,
+                             validator=FloatBoundedValidator(),
+                             doc='Thickness of the FlatPlate sample environment')
+        self.setPropertySettings('Thickness', flatPlateCondition)
+
+        self.declareProperty(name='Center', defaultValue=0.0,
+                             doc='Center of the FlatPlate sample environment')
+        self.setPropertySettings('Center', flatPlateCondition)
+
+        self.declareProperty(name='Angle', defaultValue=0.0,
+                             doc='Angle of the FlatPlate sample environment')
```

Comment:
degree? Also would be nice to be more specific about the definition of the angle. Is it with respect to the beam or beam normal? Clockwise or anti-clockwise? `-PI` to `PI`, or `0` to `2PI` ?